todo_app/app.py

```diff
@@ -1,57 +1,146 @@
+from tkinter import Variable
+import flask_login
+import requests
+import json
+import os
+import uuid
+import oauthlib.oauth2 as auth
 from todo_app.item import Item
 from todo_app.mongo_db_client import MongoDBClient
 from todo_app.view_model import ViewModel
-from flask import Flask, render_template, request, redirect, url_for
+from todo_app.user import User
+from flask import Flask, render_template, request, redirect, url_for, session
 from todo_app.flask_config import Config
-import requests
-import json
-import os
+from flask_login import login_required, LoginManager
 
 def create_app():
 
+    # App Setup
     config = Config()
     app = Flask(__name__)
     app.config.from_object(config)
     mongo_db = MongoDBClient(config.mongo_db_connection, config.mongo_db_name)
+    # Auth Setup
+    login_manager = LoginManager()
+    login_manager.init_app(app)
+
+    # when user connects - give their cookie a state - a session Variable
+
+    # check the returned state is same as session variable
+
+
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        state = str(uuid.uuid4())
+        session['state'] = state
+        url = "https://github.com/login/oauth/authorize"
+        headers = {
+            "Accept": "application/json"
+        }
+        params = {
+            "client_id": config.git_client_id,
+            "state" : state
+        }
+        response = requests.request("GET",url,headers=headers,params=params)
+        return redirect(response.url)
+
+    # Come to this route when no cookie present
+    @app.route('/login/callback', methods=['GET'])
+    def authenticate_user():
+        if (not 'state' in session) or (session['state'] != request.args['state']):
+            return render_template('unauthorised.html')
+        if "error" in request.args:
+            return render_template('unauthorised.html')
+        code = request.args['code']
```

Comment:
Nitpicking here, but ideally you should clear `session['state']` at this point so it can't be reused